File: 代码/pure-admin-thin/src/api/communicate.ts
import { http } from "@/utils/http";
import { R, baseUrlApi } from "./utils";

export class CommunicateEntity {
  id: Number;
  content: String;
  pid: Number;
  createTime: String;
  replyUserId: Number;
  children: Array<CommunicateEntity>;
}

/** list */
export const list = () => {
  return http.request<R<Array<CommunicateEntity>>>(
    "get",
    baseUrlApi("comm/list")
  );
};

/** save */
export const save = (data?: CommunicateEntity) => {
  return http.request<R<any>>("post", baseUrlApi("comm/save"), {
    data
  });
};

File: 代码/backend/src/main/java/com/fgbg/demo/entity/Communicate.java
package com.fgbg.demo.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

import lombok.Data;

/**
 * 
 * @TableName communicate
 */
@TableName(value ="communicate")
@Data
public class Communicate implements Serializable {
    /**
     * 
     */
    @TableId(type = IdType.AUTO)
    private Integer id;

    /**
     * 
     */
    private String content;

    /**
     * 
     */
    private LocalDateTime createTime;

    /**
     * 
     */
    private Integer pid;

    /**
     * 
     */
    private Integer userId;

    private Integer replyUserId;

    @TableField(exist = false)
    private List<Communicate> children;

    @TableField(exist = false)
    private static final long serialVersionUID = 1L;
}